tests/json_infra/conftest.py

```diff
@@ -175,6 +185,29 @@ def pytest_configure(config: Config) -> None:
     config.stash[desired_forks_key] = desired_forks
 
 
+def pytest_collection_modifyitems(config: Config, items: list[Item]) -> None:
```

Comment:
This hook runs after everything has been parsed and collected and then filters items out. Just thinking out loud here but wonder if we should extract the json_paths from all the entries, only parse those and then let `pytest_collection_modifyitems` filter it further.

No strong opinion on this either way though.